src/rp2_common/hardware_gpio/include/hardware/gpio.h

```diff
@@ -1428,8 +1428,8 @@ extern void gpio_debug_pins_init(void);
 #define CU_SELECT_DEBUG_PINS(x) template<> inline bool __debug_pin_settings<x>::enabled() { return true; };
 #define DEBUG_PINS_ENABLED(p) (__debug_pin_settings<p>::enabled())
 #endif
-#define DEBUG_PINS_SET(p, v) if (DEBUG_PINS_ENABLED(p)) gpio_set_mask((unsigned)(v)<<PICO_DEBUG_PIN_BASE)
-#define DEBUG_PINS_CLR(p, v) if (DEBUG_PINS_ENABLED(p)) gpio_clr_mask((unsigned)(v)<<PICO_DEBUG_PIN_BASE)
-#define DEBUG_PINS_XOR(p, v) if (DEBUG_PINS_ENABLED(p)) gpio_xor_mask((unsigned)(v)<<PICO_DEBUG_PIN_BASE)
+#define DEBUG_PINS_SET(p, v) if (DEBUG_PINS_ENABLED(p)) gpio_set_mask64((unit64_t)(v)<<PICO_DEBUG_PIN_BASE)
+#define DEBUG_PINS_CLR(p, v) if (DEBUG_PINS_ENABLED(p)) gpio_clr_mask64((unit64_t)(v)<<PICO_DEBUG_PIN_BASE)
+#define DEBUG_PINS_XOR(p, v) if (DEBUG_PINS_ENABLED(p)) gpio_xor_mask64((unit64_t)(v)<<PICO_DEBUG_PIN_BASE)
```

Comment:
Happy to move this change into a separate commit or separate PR if you'd prefer.

Reply:
yeah, I'm not sure i want to make that change - DEBUG_PINS is not really used much anyway; not sure anyone yet cares about using pins > 32

Reply:
Fair. I've removed this change, squashed and force-pushed.